src/Stratis.Bitcoin/Consensus/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -55,9 +66,57 @@ protected override async Task OnMessageReceivedAsync(INetworkPeer peer, Incoming
                 case GetProvenHeadersPayload getHeaders:
                     await this.ProcessGetHeadersAsync(peer, getHeaders).ConfigureAwait(false);
                     break;
+                case VersionPayload version:
+                    await this.ProcessVersionAsync(peer, version).ConfigureAwait(false);
+                    break;
             }
         }
 
+        /// <summary>
+        /// Processes "version" message received from the peer.
+        /// Ensures we leave some available connection slots to nodes that are enabled to serve Proven Headers, if the number of minimum PH enabled nodes connected isn't met.
+        /// </summary>
+        /// <param name="peer">Peer from which the message was received.</param>
+        /// <param name="version">Payload of "version" message to process.</param>
+        protected Task ProcessVersionAsync(INetworkPeer peer, VersionPayload version)
+        {
+            // We enforce having free slots for PH enabled nodes only when we are not in IBD.
+            if (!this.initialBlockDownloadState.IsInitialBlockDownload())
```

Comment:
I wonder if we should not reserve place for PH nodes even when in IBD @noescape00 any idea?

Reply:
We should. 
We shouldn't only before the last checkpoint.

Reply:
Did I have to change any logic here then?

Reply:
Yes you need to check if the current header is bellow checkpoint/assumed-valid, if it is then we don't need to reserve slots, if its past both of them we need to reserve slots.

Reply:
so basically what I think you need to do here is 
- First check if PH is activated
- Check if checkpoints or assumed valid (CP/AV)
- If not CP/AV enable slots reservation
- if yes CP/AV do nothing

Same logic of reservation needs to be applied when we pass the CP/AV but how do we trigger that.
Perhaps we actually need a timer (or background async loop) in this behaviour or a parent class that is shared with all this kind of behaviour types that will monitor consensus tip (CT.Tip) and trigger activation of slot reservation?